docs/config.md

```diff
@@ -663,11 +663,23 @@ java -Dspring.config.location=file:/path/to/your/myconfig.yaml -jar promregator-
 #### Option "promregator.authenticator.oauth2xsuaa.tokenServiceURL" (mandatory, if using promregator.authenticator.type=OAuth2XSUAA)
 Specifies the URL of the OAuth2 endpoint, which contains the token service of your authorization server in case of global authentication. Typically, this is the endpoint with the path `/oauth/token`, as Promregator will try to perform to establish a ["Client Credentials"-based authentication](https://www.digitalocean.com/community/tutorials/an-introduction-to-oauth-2#grant-type-client-credentials).
 
+#### Option "promregator.authenticator.oauth2xsuaa.url"
```

Comment:
I introduced `url` here instead continuing with `tokenServiceURL` since the naming in side the service key files issued by cf is `url` for client-secret based cases and `cert-url` for certificate based cases.

Of course it is also possible to continue with `tokenServiceURL` and `tokenServiceCertificateURL`.

The old `tokenServiceURL` is still fully supported for the client-secret based case.

Reply:
Argl. Things are getting now tricky:
* On the one hand "url" could be anything - for me this way too generic (you could also fill `https://github.com` in here, because that's also "some URL").
* On the other hand, there is "standard set by SAP CF" through there service key files which exactly uses this "weak" attribute name (presumably due to historical reasons).

What we need to ensure from our side at either way is that `tokenServiceURL` still works in a "compatibility" mode - we can set this attribute to "deprecated" and suggest not to use it anymore, but use `url` instead (a good example how we document deprecation can be found at `promregator.endpoint.maxProcessingTime`). We can also add a corresponding comment in the release note later to make people more aware of this change.

BTW: Using `tokenServiceURL` and then having the value only of the XSUAA server's main URL would also be wrong. So, if we wanted to properly type the URL, then the better suffix would be "xsuaaServiceURL" or so (I am open to alternative suggestions). 

Thinking aloud, here is one more idea about this: We could so this:
1. If `xsuaaServiceURL` is set, fine, take it (whatever state `tokenServiceURL` is then).
2. If `xsuaaServiceURL` is not set, but `tokenServiceURL` is, we try to strip `/oauth/token` from its end, which brings us to the value I would expect `xsuaaServiceURL` would have. If stripping fails, we raise an error (preventing startup) saying that there is a configuration mismatch, which we couldn't resolve and kindly ask the user to fix. At the same message, we can suggest to make use of the new `xsuaaServiceURL` parameter, dropping `tokenServiceURL`. 

We had similar cases already before, where he had written a "Promregator Note" in form of a Wiki page in this repo here. We then mentioned the URL of that wiki page then in the error message, guiding the user to the documentation (s)he now needs to read for details.

wdyt?

Reply:
I adjusted
- url -> xsuaaServiceURL
- cert_url -> xsuaaServiceCertURL

Reason why I started with that was: these terms (`url`, `certurl`) are used inside the service keys and also [here](https://github.com/SAP/cloud-security-xsuaa-integration/blob/fbe88ed48afd21b8f13717619770f9e06abc6cdf/java-api/src/main/java/com/sap/cloud/security/config/cf/CFConstants.java#L17) and [here](https://github.com/SAP/cloud-security-xsuaa-integration/blob/fbe88ed48afd21b8f13717619770f9e06abc6cdf/java-api/src/main/java/com/sap/cloud/security/config/cf/CFConstants.java#L40). But I don't have problems with adhering to the project conventions, of course.

The behaviour is now: when there is a `xsuaaServiceURL` already set a (deprecated) `tokenServiceURL` is not taken into account. Problem here: we depend on the order how the properties are not. Also not that cool ...
In case there is a `tokenServiceURL` which will not be ignored we strip a trailing `oauth/token`.
When we fail with getting a token we catch that exception and log it alongside with a hint pointing to the parameters which have been configured.

@eaglerainbow : Please let me know if that makes sense from your point of view. Otherwise I can adjusted again.